kernel/src/main/scala/me/rexim/morganey/ast/LambdaTerm.scala

```diff
@@ -10,11 +12,36 @@ sealed trait LambdaTerm extends MorganeyNode {
 
   def substitute(substitution : (LambdaVar, LambdaTerm)): LambdaTerm
 
-  def addBindings(context: Seq[MorganeyBinding]): LambdaTerm =
+  private def wrapBindings(context: Seq[MorganeyBinding]): LambdaTerm =
     context.foldRight(this) {
       case (MorganeyBinding(variable, value), acc) =>
         LambdaApp(LambdaFunc(variable, acc), value)
     }
+
+  private def addBindingsImpl(context: Seq[MorganeyBinding],
+                      currentVars: List[String]): Either[BindingError, LambdaTerm] = {
+      val bindings = freeVars.toList.map { x =>
+        if (!currentVars.contains(x)) {
+          context
+            .find(_.variable.name == x)
```

Comment:
Yep, I'm aware of that. I wanted to create a separate issue for that, but forgot. Thanks for reminding me!